contracts/src/protocol/OmniPortal.sol

```diff
@@ -145,8 +155,17 @@ contract OmniPortal is IOmniPortal, IOmniPortalAdmin, OwnableUpgradeable, OmniPo
             "OmniPortal: invalid proof"
         );
 
-        // update in stream block height
-        inXStreamBlockHeight[xsub.blockHeader.sourceChainId] = xsub.blockHeader.blockHeight;
+        // source chain block height of this submission
+        uint64 blockHeight = xsub.blockHeader.blockHeight;
+
+        // last seen block height for this source chain
+        uint64 lastBlockHeight = inXStreamBlockHeight[xsub.blockHeader.sourceChainId];
+
+        // update in stream block height, if it's new
+        if (blockHeight > lastBlockHeight) inXStreamBlockHeight[xsub.blockHeader.sourceChainId] = blockHeight;
```

Comment:
is this if block necessary?

Reply:
Just for partial xblock submissions to reduce gas